src/Agent.Worker/TestResults/ITestRunPublisher.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public interface ITestRunPublisher
```

Comment:
Change this to ServiceLocator Pattern. You do not have multiple implementation of this interface.
You could add a default Locator and call Get/Create Service.